UI/Panels/Device/MFShiftRegisterPanel.cs

```diff
@@ -29,52 +25,59 @@ public MFShiftRegisterPanel()
 
         public MFShiftRegisterPanel(ShiftRegister shiftRegister, List<MobiFlightPin> Pins) : this()
         {
-            ComboBoxHelper.BindMobiFlightFreePins(mfPin1ComboBox, Pins, shiftRegister.LatchPin);
-            ComboBoxHelper.BindMobiFlightFreePins(mfPin2ComboBox, Pins, shiftRegister.ClockPin);
-            ComboBoxHelper.BindMobiFlightFreePins(mfPin3ComboBox, Pins, shiftRegister.DataPin);
+            pinList = Pins; // Keep pin list stored
 
-            if (mfPin1ComboBox.Items.Count > 2)
-            {
-                mfPin1ComboBox.SelectedIndex = 0;
-                mfPin2ComboBox.SelectedIndex = 1;
-                mfPin3ComboBox.SelectedIndex = 2;                
-            }
+            this.shiftRegister = shiftRegister;
+            update_lists();
 
-            for (int i = 1; i <= MAX_MODULES; i++)
-            {
+            for (int i = 1; i <= MAX_MODULES; i++) {
                 mfNumModulesComboBox.Items.Add(i);
             }
 
-            // TODO: Complete member initialization
-            this.shiftRegister = shiftRegister;
-
-            mfPin1ComboBox.SelectedValue = byte.Parse(shiftRegister.LatchPin);
-            mfPin2ComboBox.SelectedValue = byte.Parse(shiftRegister.ClockPin);
-            mfPin3ComboBox.SelectedValue = byte.Parse(shiftRegister.DataPin);
             ComboBoxHelper.SetSelectedItem(mfNumModulesComboBox, shiftRegister.NumModules);
-            
             textBox1.Text = shiftRegister.Name;
 
             initialized = true;
         }
 
-        private void value_Changed(object sender, EventArgs e)
+        private void setNonPinValues()
         {
-            if (!initialized) return;
+            shiftRegister.Name = textBox1.Text;
+            shiftRegister.NumModules = string.IsNullOrEmpty(mfNumModulesComboBox.Text) ? "1" : mfNumModulesComboBox.Text;
+        }
+        private void update_lists()
```

Comment:
Let's change this to a more speaking name and use camel case:
UpdateFreePinsInDropDowns()